wa/commands/process.py

```diff
@@ -0,0 +1,84 @@
+#    Copyright 2014-2018 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import sys
+
+from wa import Command, settings
+from wa import discover_wa_outputs
+from wa.framework import pluginloader
+from wa.framework.version import get_wa_version
+from wa.framework.output_processor import OutputProcessor
+from wa.utils.log import log_error
+
+class ProcessCommand(Command):
+
+    name = 'process'
+    description = 'Process the output from previously run workloads.'
+
+    def __init__(self, subparsers):
+        self.loader = pluginloader
+        super(ProcessCommand, self).__init__(subparsers)
+
+    def initialize(self, context):
+        self.parser.add_argument('-d', '--process-directory', metavar='DIR', default=None,
+                                 help="""
+                                 Specify a directory to walk to find the data
+                                 from previous runs to process .If this option is
+                                 not specified, then {} will be used instead.
+                                 """.format(settings.default_output_directory))
+
+    def execute(self, config, args):
+        self.logger.debug('Version: {}'.format(get_wa_version()))
+        self.logger.debug('Command Line: {}'.format(' '.join(sys.argv)))
+
+        if args.process_directory:
+            process_directory = args.process_directory
+        else:
+            process_directory = settings.default_output_directory
+        process_directory = os.path.expandvars(process_directory)
+        self.logger.debug('Using process directory: {}'.format(process_directory))
+        if not os.path.exists(process_directory):
+            self.logger.error('Path "{}" does not exist'.format(process_directory))
+            return
+        for run_output in discover_wa_outputs(process_directory):
+            for proc in config.get_processors():
+                self.run_processor(run_output, proc)
```

Comment:
Instead of implementing your own `run_processor`, use a `ProcessorManager` here:

https://github.com/ARM-software/workload-automation/blob/next/wa/framework/output_processor.py#L35

This will ensure that the various stages for different processors get executed in the correct order.

Reply:
It appears to be the `Executor` and `Runner` in
https://github.com/ARM-software/workload-automation/blob/next/wa/framework/execution.py
that are governing the sequence rather than the `ProcessorManager`. 

Should I be trying to get this to work with the <Executor> or stick with just using the `ProcessorManager`?

The `ProcesorManager` is still useful, except that `process_job_output` requires a context to get the `job_output`, `target_info`, and `run_output` from. When called from the `Runner` this is an `ExecutionContext`.

I can see two reasonable ways around this if I'm to continue using the `ProcessorManager`, create a new class for the process command to use that contains these values, or don't use the `ProcessorManager` for that call. Do you have a preference for either of these or object to both?

Reply:
Hm, good point. It maybe worth creating a version of the `ProcessorManager` that directly takes `job_output`, `target_info`, and `run_output`; or maybe that's overkill. In any case, the important thing is to ensure that all stages are executed in the correct order, you should execute each staged for all enabled processors before moving on to the next (rather than executing all stages for each processor in turn).